src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -160,7 +175,13 @@ public FileStatus[] listStatus(Path f) throws FileNotFoundException, IOException
      */
     @Override
     public void setWorkingDirectory(Path new_dir) {
-
+        try {
+            FileSystemPathTuple tuple = generateFSPathTuple(new_dir);
+            this.workingDirectory = tuple.getPath();
+            tuple.getFileSystem().setWorkingDirectory(tuple.getPath());
+        } catch (IOException e) {
+            LOG.error("Failed setting a working directory with exception:\n{}", e.getLocalizedMessage());
```

Comment:
I don't understand: why do we want to swallow an exception here?

Reply:
The overridden method doesn't throw exceptions